lib/reek/core/smell_repository.rb

```diff
@@ -24,7 +24,8 @@ def initialize(source_description = nil, smell_types = self.class.smell_types)
       end
 
       def configure(klass, config)
-        @detectors[klass].configure_with(config) if @detectors[klass]
+        detector = @detectors[klass]
+        detector.configure_with(config) if detector
```

Comment:
Actually I'm not so sure if this guard makes sense (I know that you didnt introduce it in this PR, but it makes still sense to discuss it now).
I mean, the expectation here is that we have a valid detector. And if we don't get a valid one, we should rather raise an ArgumentError here.
WDYT? cc @chastell @mvz